File: CS-Examples/19_PivotTables/UpdateDataSource.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace UpdateDataSource
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load an excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PivotTableExample.xlsx");

            // Access the "Data" worksheet
            Worksheet data = workbook.Worksheets["Data"];

            // Modify the data source by changing the value in cell A2 to "NewValue"
            data.Range["A2"].Text = "NewValue";

            // Modify the data source by changing the value in cell D2 to 28000
            data.Range["D2"].NumberValue = 28000;

            // Access the worksheet containing the pivot table
            Worksheet sheet = workbook.Worksheets["PivotTable"];

            // Get the first pivot table from the worksheet
            XlsPivotTable pt = sheet.PivotTables[0] as XlsPivotTable;

            // Set the pivot table's cache to refresh on load
            pt.Cache.IsRefreshOnLoad = true;

            // Calculate and update the pivot table data
            pt.CalculateData();

            // Specify the filename for the updated workbook
            String result = "UpdateDataSource_result.xlsx";

            // Save the modified workbook to a file (in Excel 2010 format)
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/19_PivotTables/UpdateDataSource.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace UpdateDataSource
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "PivotTableExample.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PivotTableExample.xlsx")

            ' Get the worksheet named "Data" from the workbook
            Dim data As Worksheet = workbook.Worksheets("Data")

            ' Set the value of cell A2 in the "Data" worksheet to "NewValue"
            data.Range("A2").Text = "NewValue"

            ' Set the numeric value of cell D2 in the "Data" worksheet to 28000
            data.Range("D2").NumberValue = 28000

            ' Get the worksheet named "PivotTable" from the workbook
            Dim sheet As Worksheet = workbook.Worksheets("PivotTable")

            ' Get the first PivotTable from the "PivotTable" worksheet
            Dim pt As XlsPivotTable = TryCast(sheet.PivotTables(0), XlsPivotTable)

            ' Enable automatic refresh of the PivotTable cache on load
            pt.Cache.IsRefreshOnLoad = True

            ' Calculate the data in the PivotTable
            pt.CalculateData()

            ' Specify the name for the resulting file as "UpdateDataSource_result.xlsx"
            Dim result As String = "UpdateDataSource_result.xlsx"

            ' Save the modified workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
